nextflow/utils/utils.nf

```diff
@@ -60,3 +60,25 @@ def print_summary() {
     )
   }
 }
+
+def check_JVM_mem (min=0.4) {
+  /*
+    Check memory available to the Java Virtual Machine (JVM) instance running
+    the Nextflow head node
+
+    Throws error if memory is lower than parameter min
+
+    @param min Min memory in GiB (default: 0.4)
+  */
+
+  mem = Runtime.getRuntime().maxMemory() / (1024 ** 3) // in GiB
+  if (mem < min) {
+    log.error """
+    ERROR: The memory of the Nextflow head node (${mem.round(2)} GiB) is below the recommended (${min} GiB); please try the following actions:
+      - Increase the JVM's max RAM percentage: export NXF_JVM_ARGS="-XX:InitialRAMPercentage=25 -XX:MaxRAMPercentage=75"
+      - Increase the JVM heap size memory: export NXF_OPTS="-Xms50m -Xmx${min}g"
+      - Increase the memory of the job used to run the Nextflow pipeline: salloc --time 6:00:00 --mem ${min * 4}GB
```

Comment:
salloc --mem seems to only take whole numbers

Reply:
There is also a class for [memory unit](https://www.nextflow.io/docs/latest/script.html#memoryunit):

Reply:
Unfortunately, the memory unit class automatically rounds down the values, so I don't think it is useful in this context (if only it would round up...)

I rounded up the memory values in the commands to work as expected. Thanks @likhitha-surapaneni!